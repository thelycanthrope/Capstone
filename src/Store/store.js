import {createStore,applyMiddleware} from 'redux'
import productReducer from '../Actions/productReducer'
import logger from 'redux-logger'
import thunk from 'redux-thunk'

const store=createStore(productReducer, applyMiddleware(logger,thunk))

export default store