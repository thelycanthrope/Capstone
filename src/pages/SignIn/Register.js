import { Formik,Form,Field,ErrorMessage, useFormikContext } from 'formik'
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import * as Yup from 'yup'
import { fetchUser } from '../../Actions/actions'
import {toast} from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'
import addUser from '../../Actions/addUser'
import { Prompt, useHistory } from 'react-router'
import NavBar from '../../Components/NavBar'



toast.configure();

const initialValues={
  email:"",
  password:"",
  fname:"",
  lname:"",
    location:"",
    contactno:""
}

const validationSchema = Yup.object({
    email:Yup.string().email('Invalid format').required('Email is mandatory field'),
	password:Yup.string().required('password is mandatory field'),
	fname:Yup.string().required('First Name is mandatory field'),
	lname:Yup.string().required('Last Name is mandatory field') ,
    location:Yup.string().required('Location is mandatory field'),
    contactno:Yup.string().required('Contact No. is mandatory field')
})


const PromptIfDirty = () => {
  const formik = useFormikContext();
  return (
    <Prompt
      when={formik.dirty && formik.submitCount === 0}
      message="Are you sure you want to leave? You have with unsaved changes."
    />
  );
};



function Register(props) {
  const inusers=useSelector( (state) => {return state});
  const history=useHistory()
  const dispatch=useDispatch();
  // const formik = useFormikContext();
  useEffect( () => {
    dispatch(fetchUser())
  },[dispatch])
  let users;
  const onsubmit = (values,actions) => {
  
    let i;
    let ret=false;
    for ( i = 0; i < users.length; i++) {
      if (values.email === users[i].email) {
      ret=true;
      break;
      }
    }
    if(!ret){
      console.log(values,'ret');
      addUser(values);
      actions.resetForm({values:''})
      toast.success('Registration successfull ...sign in to do opertions',{position:toast.POSITION.TOP_CENTER})
      history.push('/signin')
    }
    else{
      toast.error('User Exists with this email',{position:toast.POSITION.TOP_CENTER})
    }
    
  }
  
  (typeof(inusers)!=='undefined')?users=[inusers.users]:console.log();
 


    return (
      <div className="bg-dark">
        <NavBar/>
        <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onsubmit}>
    
            <Form className=" col-sm "><PromptIfDirty/>
          <h3 className="text-white container text-center pt-5">New user?  fill the details to Register</h3>
            <div className="mx-5 px-5 py-5 border">
              <div >
                <label htmlFor="Email1" className="form-label text-white">Email address</label>
                <div style={{color:"red"}}><ErrorMessage name="email"/></div>
                <Field type="email" className="form-control bg-dark text-white" placeholder="email address" id="Email1" aria-describedby="emailHelp"  name="email"/>
              </div>

              <div >
                <label htmlFor="Password1" className="form-label  text-white mt-3">Password</label>
                <div style={{color:"red"}}><ErrorMessage name="password"/></div>
                <Field type="password" className="form-control bg-dark text-white" placeholder="Password" id="Password1"  name="password"/>
              </div>

              <div className="row">

              <div className="col-sm">
                <label htmlFor="userfname" className="form-label text-white mt-3">First Name</label>
                <div style={{color:"red"}}><ErrorMessage name="fname"/></div>
                <Field type="text" className="form-control bg-dark text-white" placeholder="First Name" id="userfname" aria-describedby="emailHelp" name="fname"/>
                </div>
                <div className="col-sm">
                <label htmlFor="uselname" className="form-label text-white mt-3">Last Name</label>
                <div style={{color:"red"}}><ErrorMessage name="lname"/></div>
                <Field type="text" className="form-control bg-dark text-white" placeholder="Last Name" id="uselname" aria-describedby="emailHelp" name="lname"/>
              </div>
              </div>
              <div className="row">

              <div className="col-sm">
                <label htmlFor="location" className="form-label text-white mt-3">Location</label>
                <div style={{color:"red"}}><ErrorMessage name="location"/></div>
                <Field type="text" className="form-control bg-dark text-white" placeholder="location" id="location" aria-describedby="emailHelp" name="location"/>
                </div>
                <div className="col-sm">
                <label htmlFor="contact" className="form-label text-white mt-3">Contact No.</label>
                <div style={{color:"red"}}><ErrorMessage name="contactno"/></div>
                <Field type="text" className="form-control bg-dark text-white" placeholder="contact No." id="contact" aria-describedby="emailHelp" name="contactno"/>
              </div>
              </div>

              <button type="submit" className="btn bg-dark border-white text-white w-100 mt-3">Register</button>
            </div>
            <br/>
            <br/>
            <br/>
           
          </Form>
         

        </Formik>
        </div>
    )
}




export default Register
