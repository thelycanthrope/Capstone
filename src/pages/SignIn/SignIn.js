import React, { useEffect } from 'react'
import './signin.css'
import { Formik,Form,Field,ErrorMessage } from 'formik'
import * as Yup from 'yup'
import { Link } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'
import { fetchUser} from '../../Actions/actions'
import {toast} from 'react-toastify'
import NavBar from '../../Components/NavBar'




toast.configure();
const initialValues={
  email:"",
  password:""
}
const validationSchema =Yup.object ({
  email:Yup.string().email('Invalid Format').required('Mandatory field'),
  password:Yup.string().required('Mandatory Field') 
})



function SignIn(props) {


  const onsubmit = (values,actions) =>{
    
   
    let i=0;
    let c=false;

    for(i;i<se.users.length;i++)
    {
      if(values.email===se.users[i].email){
        //email matches
        c=true;
        if(values.password===se.users[i].password){
          //passwrod matches
          
          // signIn(values.email)//fires sign in action

          localStorage.setItem('loggedin',values.email)

          toast.success('Credentials mathes...signed in successfully',{position:toast.POSITION.TOP_CENTER})
          props.history.push('/home')
          
          // window.location.reload()
          break;
        }
        else{
         //wrong password
          toast.error('Wrong Password!',{position:toast.POSITION.TOP_CENTER})
          break;
        }
      }
      else{
        //wrong email
        c=false;
      }
    }

    //wrong email
   if(c===false){
    
     toast.error('wrong mail',{position:toast.POSITION.TOP_CENTER})
   }
  
   

  }
  const dispatch=useDispatch();
  const se=useSelector( state => state)
  useEffect ( ( ) =>{
    dispatch(fetchUser());
  },[dispatch])

  
  
console.log('se',se);

    return (
      <div className="bg-dark ">
       <NavBar/>
        <br></br>
        <div className="row">
          
        <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onsubmit}>

          {/* ***********************signin form************************** */}
          <Form className="container col-sm " id="here2">
            <h3 className="text-white mx-5 "> Sign In</h3>
            <div className="mx-5 px-5 my-5 py-5 border " id="here">
              <div className="form-group ">
                <label htmlFor="exampleInputEmail1" className="text-white">Email address</label>
                <div  className="text-danger"><ErrorMessage name="email"/></div>
                <Field type="email" className="form-control  bg-dark text-white" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email"   name="email"/>
                {/* <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small> */}
              </div>
              <div className="form-group ">
                <label htmlFor="exampleInputPassword1" className="text-white">Password</label>
                <div className="text-danger"><ErrorMessage name="password"/></div>
                <Field type="password" className="form-control bg-dark text-white" id="exampleInputPassword1" placeholder="Password" name="password" />
              </div>
              <button type="submit" className="btn bg-dark border-white text-white">SignIn</button>
            </div>
          </Form>

                  {/************registration form********************* */  }
          
          </Formik>
          {/* <Register/> */}
          <div className="col-sm border mx-5 mb-3">
            <h1 className="text-white text-center mt-5 py-5">Hello Friend!</h1>
            <h5 className="text-white text-center " style={{fontFamily:'"Times New Roman", Times, cursive'}}>Fill your personal details to start journey with us!</h5>
              <br></br> <br></br>
            <Link to='/register' > <div className="text-center"><button className=" btn btn-dark bg-dark text-white border"> Register</button> </div> </Link>
            </div> 
        </div>
        
          <br/>
          <br/>
          <br/>
          <br/>
          <br/>
          <br/>
          <br/>
      </div>
    )
}

export default SignIn
