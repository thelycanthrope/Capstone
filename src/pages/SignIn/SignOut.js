import React from 'react'
import { toast } from 'react-toastify'


function SignOut() {
    return (
        <div>
            {toast.success('Signedout successfully')}
        </div>
    )
}

export default SignOut
