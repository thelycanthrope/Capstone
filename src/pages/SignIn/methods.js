import axios from "axios"


export const signIn = (mail) => {
    
    axios.put('http://localhost:5500/loggedin/1',{loggedin:"true",email:mail})
    .then( r => console.log(r))
    .catch( e =>console.log(e))
    
}


export const signOut = () => {
    axios.put('http://localhost:5500/loggedin/1',{loggedin:"false"})
    .then( r => console.log(r))
    .catch( e =>console.log(e)) 
    
}

