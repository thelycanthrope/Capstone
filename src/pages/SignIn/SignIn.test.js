import React from 'react';

import {shallow,configure,mount} from 'enzyme';
import SignIn from './SignIn';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../Store/store'

import configureStore from 'redux-mock-store';


configure({ adapter: new Adapter() });
const mockStore = configureStore([]);





describe('First SignIn Test', () => {
    let wrapper;
    let store;
    beforeEach(() => {
        store = mockStore({
            myState: 'sample text',
        });
        wrapper = shallow(<Provider store={store}><SignIn /></Provider>)
    })

    it('should take snap', () => {
        let comp = shallow(<BrowserRouter><Provider store={store}><SignIn /></Provider></BrowserRouter>).dive().render()
        expect(toJson(comp)).toMatchSnapshot()
    })

    it('should have 2 input', () => {
        let comp = shallow(<BrowserRouter><Provider store={store}><SignIn /></Provider></BrowserRouter>).dive().render()
        expect(comp.find('input').length).toEqual(2)
    })

    it('input email test', () => {
        let comp = shallow(<BrowserRouter><Provider store={store}><SignIn /></Provider></BrowserRouter>).dive().render()
        expect(comp.find('label').length).toEqual(2)
    })

})
