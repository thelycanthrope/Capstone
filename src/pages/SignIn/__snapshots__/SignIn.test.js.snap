// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First SignIn Test should take snap 1`] = `
<div
  class="bg-dark "
>
  <br />
  <div
    class="row"
  >
    <form
      action="#"
      class="container col-sm "
      id="here2"
    >
      <h3
        class="text-white mx-5 "
      >
         Sign In
      </h3>
      <div
        class="mx-5 px-5 my-5 py-5 border "
        id="here"
      >
        <div
          class="form-group "
        >
          <label
            class="text-white"
            for="exampleInputEmail1"
          >
            Email address
          </label>
          <div
            class="text-danger"
          />
          <input
            aria-describedby="emailHelp"
            class="form-control  bg-dark text-white"
            id="exampleInputEmail1"
            name="email"
            placeholder="Enter email"
            type="email"
            value=""
          />
        </div>
        <div
          class="form-group "
        >
          <label
            class="text-white"
            for="exampleInputPassword1"
          >
            Password
          </label>
          <div
            class="text-danger"
          />
          <input
            class="form-control bg-dark text-white"
            id="exampleInputPassword1"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <button
          class="btn bg-dark border-white text-white"
          type="submit"
        >
          SignIn
        </button>
      </div>
    </form>
    <div
      class="col-sm border mx-5 mb-3"
    >
      <h1
        class="text-white text-center mt-5 py-5"
      >
        Hello Friend!
      </h1>
      <h5
        class="text-white text-center "
        style="font-family:\\"Times New Roman\\", Times, cursive"
      >
        Fill your personal details to start journey with us!
      </h5>
      <br />
       
      <br />
      <a
        href="/register"
      >
         
        <div
          class="text-center"
        >
          <button
            class=" btn btn-dark bg-dark text-white border"
          >
             Register
          </button>
           
        </div>
         
      </a>
    </div>
  </div>
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
</div>
`;
